test/Microsoft.Azure.Devices.Edge.Test.Common/IPlatform.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Test.Common.Certs;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public interface IPlatform
+    {
+        Task<string> CollectDaemonLogsAsync(DateTime testStartTime, string filePrefix, CancellationToken token);
+
+        IEdgeDaemon CreateEdgeDaemon(Option<string> installerPath);
+
+        // After calling this function, the following files will be available under {scriptPath}:
+        //  certs/iot-edge-device-{deviceId}-full-chain.cert.pem
+        //  private/iot-edge-device-{deviceId}.key.pem
+        Task<EdgeCertificates> GenerateEdgeCertificatesAsync(string deviceId, string scriptPath, CancellationToken token);
+
+        // After calling this function, the following files will be available under {scriptPath}:
+        //  certs/iot-device-{leafDeviceId}-full-chain.cert.pem
+        //  private/iot-device-{leafDeviceId}.key.pem
+        Task<LeafCertificates> GenerateLeafCertificatesAsync(string leafDeviceId, string scriptPath, CancellationToken token);
+
+        StoreName GetCertificateStoreName();
+
+        string GetConfigYamlPath();
```

Comment:
should this method be better in EdgeDaemon class?

Reply:
It could be, but I don't really think it's better.

Reply:
Here's what I can do for `GetConfigYamlPath`. I'll get rid of the function altogether, and just pass that path value into the `DaemonConfiguration` constructor, since that's the only caller.

Reply:
Done.